src/main.rs

```diff
@@ -1,27 +1,68 @@
-#[allow(unused_imports)]
-use std::net::UdpSocket;
+mod answer;
+mod dns_types;
+mod header;
+mod message;
+mod name;
+mod question;
+
+use message::DNSMessage;
+
+use crate::answer::DNSAnswer;
+use crate::header::DNSHeader;
+use std::{env, net::UdpSocket};
 
 fn main() {
-    // You can use print statements as follows for debugging, they'll be visible when running tests.
-    println!("Logs from your program will appear here!");
+    let args: Vec<String> = env::args().collect();
+    let resolver: Option<String> = if args.len() > 1 && args[1] == "--resolver" {
+        Some(args[2].clone())
+    } else {
+        None
+    };
```

Comment:
IHMO it would be more idomatic and exensible to have a `Cli` struct that defines and documents the interface of the program. It would implement `FromStr` (or `TryFrom<&str>`) and provide a `from_env` method that calls `std::env::args_os` and returns either the arguments or exists the programs returning an help or pretty error message

Reply:
It could include the `listen_addr` and `bind_addr` with default to the constants used here